validator/jit_machine.go

```diff
@@ -196,6 +196,22 @@ func (machine *JitMachine) prove(
 			return state, err
 		}
 	}
+
+	// send known preimages
+	knownPreimages := make(map[[32]byte][]byte)
```

Comment:
This should probably be `entry.Preimages`

Reply:
@tsahee do we still want this? When you were first starting your work I was hoping it'd solve a minor thing you'd need in the very beginning. I'm not sure what you're doing now regarding validation though :)

Reply:
Oops.. I thought it was already merge :D I think best would be to merge it to my branch now.

Reply:
I'm going to mark this PR as a draft since we aren't planning on merging it into master